src/main/scala/nl/biopet/utils/ngs/package.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2014 Biopet
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package nl.biopet.utils
+
+package object ngs {
+
+  /**
+    * This method convert the sequence to a 2 bit encoding Integer
+    * Size of the sequence of maximum 16 nucleotides
+    * @param sequence sequence as string, max length is 16
+    * @return
+    */
+  def sequenceTo2bitSingleInt(sequence: String): Int = {
+    sequenceTo2bitSingleInt(sequence.getBytes)
+  }
+
+  val A: Byte = 'A'.toByte
+  val T: Byte = 'T'.toByte
+  val C: Byte = 'C'.toByte
+  val G: Byte = 'G'.toByte
+  val a: Byte = 'a'.toByte
+  val t: Byte = 't'.toByte
+  val c: Byte = 'c'.toByte
+  val g: Byte = 'g'.toByte
+
+  /**
+    * This method convert the sequence to a 2 bit encoding Integer
+    * Size of the sequence of maximum 16 nucleotides
+    * If the length is smaller then 16 the rest is filled with A
+    * @param sequence sequence as Array[Byte], max length is 16
+    * @return
+    */
+  def sequenceTo2bitSingleInt(sequence: Array[Byte]): Int = {
+    require(sequence.length <= 16, "Sequence is to long for a single Int")
+
+    sequence.zipWithIndex.map {
+      case (nuc, idx) =>
```

Comment:
Not clear to me what idx means. If it's an index: what sort of index?

Reply:
idx is the position inside the sequence (0-bases)